Sources/Widgets/Core/WidgetManager/index.js

```diff
@@ -288,38 +298,18 @@ function vtkWidgetManager(publicAPI, model) {
     }
   }
 
-  const handleEvent = (eventName) => {
-    let guard = false;
-    return (callData) => {
-      if (
-        guard ||
-        model.isAnimating ||
-        !model.pickingEnabled ||
-        model._selectionInProgress
-      ) {
-        return macro.VOID;
-      }
-
-      const updatePromise = updateSelection(callData);
-      macro.measurePromiseExecution(updatePromise, (elapsed) => {
-        // 100ms is deemed fast enough. Anything higher can degrade usability.
-        if (elapsed > 100) {
-          macro.vtkWarningMacro(
-            `vtkWidgetManager updateSelection() took ${elapsed}ms on ${eventName}`
-          );
-        }
-      });
+  const handleEvent = async (callData, fromTouchEvent = false) => {
+    if (
+      model.isAnimating ||
+      !model.pickingEnabled ||
+      model._selectionInProgress
```

Comment:
I'm not sure events should be discarded when processing one event. I guess it's important to handle the last event (debounce ? throttle?). 
Alternatively, we could discard only the events of the same type as the one being `in progress.

Reply:
Good point. I toyed with the idea of handling the last event, discarding the results of previous events, but that didn't make it in this PR. I will re-visit that idea.